### DISCUSSION TITLE
Azure Function Sample, but with dynamic site url

### DISCUSSION THREAD

**Discussion by RichieRoxx** [pnp/pnpcore#733] [Q&A] [answered]
Hey everyone,

I've implemented an azure function based on this sample (https://github.com/pnp/pnpcore/blob/dev/samples/Demo.AzureFunction.OutOfProcess.AppOnly/Program.cs) but in my version i can't get the siteurl from the environmentvariables, but need it to be dynamic. The site I will work with will always be dynamic and be in the service bus message that triggers my function. I wonder what approach I should take to implement that. Any ideas and help/examples are appreciated!

Cheers

Richard

**Comment by RichieRoxx** [comment, marked as answer]
ok, was actually able to answer it myself, so I'll share this for anyone else maybe needing the same:


instead of of


```
using (var pnpContext = await ContextFactory.CreateAsync("Default"))
{
//doing sharepointstuff here
}

```
use 
```
using (var pnpContext = await ContextFactory.CreateAsync(SiteUri))
{
//doing sharepointstuff here
}
```

SiteUri needs to be of type Uri of course and you need to define a default authentication provider in program.cs/startup.cs
```
 // Add and configure PnP Core SDK
                    services.AddPnPCore(options =>
                    {
//Begin of lines that need to be added to the referenced sample above                       
 // Configure an authentication provider with certificate (Required for app only)
                        var authProvider = new X509CertificateAuthenticationProvider(azureFunctionSettings.ClientId,
                        azureFunctionSettings.TenantId,
                        StoreName.My,
                        StoreLocation.CurrentUser,
                        azureFunctionSettings.CertificateThumbprint);
                        // And set it as default
                        options.DefaultAuthenticationProvider = authProvider;
//End of lines that need to be added to the referenced sample above
                        // Add the base site url
                        options.Sites.Add("Default", new PnPCoreSiteOptions
                        {
                            SiteUrl = azureFunctionSettings.SiteUrl
                        });
                    });
```

some of the lines from the sample where the "Default"-Service is created are now obviously redundant...

still any feedback highly welcome if you have tipps or additions!

**Reply by jansenbe** [reply]
The minimal console sample (https://github.com/pnp/pnpcore/blob/dev/samples/Demo.Console.Minimal/Program.cs) also shows directly using an URI.

**Reply by RichieRoxx** [reply]
its where I got it from :) thanks!
